test/e2e/specs/editor/blocks/navigation.spec.js

```diff
@@ -317,3 +317,480 @@ test.describe( 'Navigation block', () => {
+					hasText: 'Block 1 of 1, Level 2', // proxy for filtering by description.
+				} )
+				.getByText( 'Test Submenu Item' )
+		).toBeVisible();
+	} );
+
+	test( `can add new menu items`, async ( {
+		admin,
+		page,
+		editor,
+		requestUtils,
+	} ) => {
+		await admin.createNewPost();
+		await requestUtils.createNavigationMenu( {
+			title: 'Test Menu',
+			content: `<!-- wp:navigation-link {"label":"Top Level Item 1","type":"page","id":250,"url":"http://localhost:8888/quod-error-esse-nemo-corporis-rerum-repellendus/","kind":"post-type"} /-->
+			<!-- wp:navigation-submenu {"label":"Top Level Item 2","type":"page","id":250,"url":"http://localhost:8888/quod-error-esse-nemo-corporis-rerum-repellendus/","kind":"post-type"} -->
+				<!-- wp:navigation-link {"label":"Test Submenu Item","type":"page","id":270,"url":"http://localhost:8888/et-aspernatur-recusandae-non-sint/","kind":"post-type"} /-->
+			<!-- /wp:navigation-submenu -->`,
+		} );
+
+		await editor.insertBlock( { name: 'core/navigation' } );
+
+		await editor.openDocumentSettingsSidebar();
+
+		const listViewTab = page.getByRole( 'tab', {
+			name: 'List View',
+		} );
+
+		await listViewTab.click();
+
+		const listView = page.getByRole( 'treegrid', {
+			name: 'Block navigation structure',
+			description: 'Structure for navigation menu: Test Menu',
+		} );
+
+		const appender = listView.getByRole( 'button', {
+			name: 'Add block',
+		} );
+
+		await expect( appender ).toBeVisible();
+
+		await appender.click();
+
+		// Expect to see the block inserter.
+		await expect(
+			page.getByRole( 'searchbox', {
+				name: 'Search for blocks and patterns',
+			} )
+		).toBeFocused();
+
+		const blockResults = page.getByRole( 'listbox', {
+			name: 'Blocks',
+		} );
+
+		await expect( blockResults ).toBeVisible();
+
+		const blockResultOptions = blockResults.getByRole( 'option' );
+
+		// Expect to see the Page Link and Custom Link blocks as the nth(0) and nth(1) results.
+		// This is important for usability as the Page Link block is the most likely to be used.
+		await expect( blockResultOptions.nth( 0 ) ).toHaveText( 'Page Link' );
+		await expect( blockResultOptions.nth( 1 ) ).toHaveText( 'Custom Link' );
+
+		// Select the Page Link option.
+		const pageLinkResult = blockResultOptions.nth( 0 );
+		await pageLinkResult.click();
+
+		// Expect to see the Link creation UI be focused.
+		const linkUIInput = page.getByRole( 'combobox', {
+			name: 'URL',
+		} );
+		await expect( linkUIInput ).toBeFocused();
+
+		const linkUIResults = page.getByRole( 'listbox', {
+			name: 'Recently updated',
+		} );
+
+		await expect( linkUIResults ).toBeVisible();
+
+		// click on the first result
+		const firstResult = linkUIResults.getByRole( 'option' ).nth( 0 );
+
+		// Grab the text from the first result so we can check it was inserted.
+		const firstResultText = await firstResult
+			.locator( '.block-editor-link-control__search-item-title' ) // this is the only way to get the label text without the URL.
```

Comment:
I tried to find a different way to target this to [avoid the CSS selector](https://playwright.dev/docs/best-practices#prefer-user-facing-attributes-to-xpath-or-css-selectors) but I couldn't :S

Reply:
Nope. There isn't a way which probably points to a flaw in the markup of the component. It's due an a11y review as part of https://github.com/WordPress/gutenberg/issues/49091

Reply:
maybe add a comment referring that PR so we can update the test once that gets merged?

Reply:
I would agree with @MaggieCabrera to add a comment referencing that PR, but it's non-blocking 👍 .

Reply:
> Have the test select a known page title, rather than trying to read the text from the option.

That might be better.

Reply:
Only thing is if we ever change the implementation of "Recently Added" results in Link UI the test would start failing.

That's why I wanted to get the first result's text. I can extract to a helper on the utility class if that helps.

Reply:
I added a `getSearchResultText` method to the `LinkControl` test util class.